drivers/vulkan/vulkan_context.h

```diff
@@ -154,7 +159,7 @@ class VulkanContext {
 		VkSwapchainKHR swapchain = VK_NULL_HANDLE;
 		SwapchainImageResources *swapchain_image_resources = VK_NULL_HANDLE;
 		VkPresentModeKHR presentMode = VK_PRESENT_MODE_FIFO_KHR;
-		VkSemaphore image_acquired_semaphores[FRAME_LAG];
+		VkSemaphore image_acquired_semaphores[MAX_FRAME_LAG] = {};
```

Comment:
Could you elaborate why multiple acquired semaphores are needed per window? 
AFAIK, only one swapchain image is generally acquired at a time. (this is actually an issue I was looking into for https://github.com/godotengine/godot/issues/84137 and in fact, I saw (in superficial testing) no detriment in keeping just 1 image_aquired_semaphore per Window.

(I'm quite possibly bad and wrong about this)

Reply:
Hi!

I had the same question actually!

Here's the thing:

1. Indeed just one semaphore for all windows is how I think should be properly done (unless you're careful with barriers and want to explicitly start doing Compute Work using async compute for the next frame while the current frame is still being worked on)
2. In OgreNext we use one semaphore and funny enough, that triggered driver bugs for various drivers (which have been long been fixed)
3. Official and unofficial samples use one semaphore per swapchain (this is most likely why the drivers didn't catch that bug for OgreNext...), and since I saw that Godot was doing the same, I decided not to change this.

But yes, for the record for desktop\*, I am more comfortable with one semaphore for everything because that's how I think proper synchronization should be.

\* I say desktop because I'm having timing issues on Android that I can't fix in OgreNext and I never tested the theory of whether they're caused because of having one semaphore for everything. But on the other hand, Android has a lot of problems of all sorts.